### ISSUE TITLE
graph doesn't show all links to nodes

### ISSUE THREAD

**Issue by jcolson** [obsidian-html/obsidian-html#40]
I'm not sure how to report on this, but I've noticed that there are node links that should be there (they are in obsidian for example) but they are not in the html output of the graph.

if there is any debug that I can help with, let me know.

**Comment by dwrolvink** [comment]
If this happens when process_all: True, then it is #22 
If it happens when that value is false then that would be a new bug. 

Basically a file is not processed twice, and that causes links to not be created when they should. See #22 for more info.

**Comment by jcolson** [comment]
> If this happens when process_all: True, then it is #22 If it happens when that value is false then that would be a new bug.
> 
> Basically a file is not processed twice, and that causes links to not be created when they should. See #22 for more info.

I haven't tried setting `process_all: True`, so I would assume this is a new bug.

**Comment by jcolson** [comment]
![image](https://user-images.githubusercontent.com/2362647/152686024-bdb422cc-95cc-469a-afb4-d80f93160365.png)
vs
<img width="433" alt="image" src="https://user-images.githubusercontent.com/2362647/152686032-45712359-8bca-410c-9d4b-26e391af3b88.png">

Note that there should be several nodes linked to Jolo, but obsidian-html only shows one

**Comment by dwrolvink** [comment]
Oh I just remembered I noticed this yesterday too. The proposed fix in #22 should fix this too.

See the lines:
Note: this will also fix another bug:
- Links missing in graph view more than one note links to another (processed note does not get processed again).

The problem you're seeing is that session links to all those notes that Jolo Brekk links to. But Session will call those notes, and set the graph links for them, which prevents Jolo Brekk from linking to them, because they are already processed.

The proposed fix will also make a full bidirectional graph json, which can be used later to add backlink support.
